src/CommunityToolkit.Maui/Converters/BoolToObjectConverter.shared.cs

```diff
@@ -51,7 +52,8 @@ public class BoolToObjectConverter<TObject> : ValueConverterExtension, IValueCon
 	/// <param name="parameter">Additional parameter for the converter to handle. This is not implemented.</param>
 	/// <param name="culture">The culture to use in the converter.  This is not implemented.</param>
 	/// <returns>True if value equals <see cref="TrueObject"/>, otherwise False.</returns>
-	public object ConvertBack(object? value, Type? targetType, object? parameter, CultureInfo? culture)
+	[return: NotNull]
+	public object? ConvertBack(object? value, Type? targetType, object? parameter, CultureInfo? culture)
```

Comment:
I am just curious, do we have generic converters?
`IConverter<TInput,TOutput>`
Where we can specify constraints TInput:notnull

Reply:
We do not. 

I like that idea, but I'm hesitant to do that until the MAUI team enables nullable on `IValueConverter` because I honestly don't know which way the team will go: will all parameters be nullable?; will all parameters be not nullable?; will some parameters be nullable?; will the return types be nullable?; etc. 

If we guess wrong, we would end up with a lot of re-work and breaking changes on our APIs. 

With `ICommunityToolkitValueConverter`, because we assume everything in `IValueConverter` is nullable, there won't be any breaking changes to our users if/when the team enables nullable. For example, if the MAUI team decides that `Type targetType` doesn't need to be nullable, there's not breaking changes for our users since we're already handling the null check.

I love this for the future, though! Want to open that as a Discussion for a future Proposal?

Reply:
Yes, let's move it to discussion